Source/Core/VideoCommon/BPMemory.h

```diff
@@ -354,169 +469,384 @@ struct TevStageCombiner
+                     scale.ss1, 1.f / (1 << scale.ss1), scale.ts1, 1.f / (1 << scale.ts1));
+  }
+};
 
 union RAS1_IREF
 {
-  struct
-  {
-    u32 bi0 : 3;  // Indirect tex stage 0 ntexmap
-    u32 bc0 : 3;  // Indirect tex stage 0 ntexcoord
-    u32 bi1 : 3;
-    u32 bc1 : 3;
-    u32 bi2 : 3;
-    u32 bc3 : 3;
-    u32 bi4 : 3;
-    u32 bc4 : 3;
-    u32 rid : 8;
-  };
+  BitField<0, 3, u32> bi0;  // Indirect tex stage 0 ntexmap
+  BitField<3, 3, u32> bc0;  // Indirect tex stage 0 ntexmap
+  BitField<6, 3, u32> bi1;
+  BitField<9, 3, u32> bc1;
+  BitField<12, 3, u32> bi2;
+  BitField<15, 3, u32> bc3;  // Typo?
+  BitField<18, 3, u32> bi4;
+  BitField<21, 3, u32> bc4;
   u32 hex;
 
   u32 getTexCoord(int i) const { return (hex >> (6 * i + 3)) & 7; }
   u32 getTexMap(int i) const { return (hex >> (6 * i)) & 7; }
 };
+template <>
+struct fmt::formatter<RAS1_IREF>
+{
+  constexpr auto parse(format_parse_context& ctx) { return ctx.begin(); }
+  template <typename FormatContext>
+  auto format(const RAS1_IREF& indref, FormatContext& ctx)
+  {
+    // The field names here are suspicious, since there is no bi3 or bc2
+    return format_to(ctx.out(),
+                     "Stage 0 ntexmap: {}\nStage 0 ntexcoord: {}\n"
+                     "Stage 1 ntexmap: {}\nStage 1 ntexcoord: {}\n"
+                     "Stage 2 ntexmap: {}\nStage 2 ntexcoord: {}\n"
+                     "Stage 3 ntexmap: {}\nStage 3 ntexcoord: {}",
+                     indref.bi0, indref.bc0, indref.bi1, indref.bc1, indref.bi2, indref.bc3,
+                     indref.bi4, indref.bc4);
+  }
+};
 
 // Texture structs
+enum class WrapMode : u32
+{
+  Clamp = 0,
+  Repeat = 1,
+  Mirror = 2,
+};
+template <>
+struct fmt::formatter<WrapMode> : EnumFormatter<WrapMode::Mirror>
+{
+  formatter() : EnumFormatter({"Clamp", "Repeat", "Mirror"}) {}
+};
+
+enum class MipMode : u32
+{
+  None = 0,
+  Point = 1,
+  Linear = 2,
+};
+template <>
+struct fmt::formatter<MipMode> : EnumFormatter<MipMode::Linear>
+{
+  formatter() : EnumFormatter({"None", "Mip point", "Mip linear"}) {}
+};
+
+enum class FilterMode : u32
+{
+  Near = 0,
+  Linear = 1,
+};
+template <>
+struct fmt::formatter<FilterMode> : EnumFormatter<FilterMode::Linear>
+{
+  formatter() : EnumFormatter({"Near", "Linear"}) {}
+};
+
+enum class LODType : u32
+{
+  Edge = 0,
+  Diagonal = 1,
+};
+template <>
+struct fmt::formatter<LODType> : EnumFormatter<LODType::Diagonal>
+{
+  formatter() : EnumFormatter({"Edge LOD", "Diagonal LOD"}) {}
+};
+
+enum class MaxAnsio
+{
+  One = 0,
+  Two = 1,
+  Four = 2,
+};
+template <>
+struct fmt::formatter<MaxAnsio> : EnumFormatter<MaxAnsio::Four>
+{
+  formatter() : EnumFormatter({"1", "2 (requires edge LOD)", "4 (requires edge LOD)"}) {}
+};
 
 union TexMode0
 {
-  enum TextureFilter : u32
+  BitField<0, 2, WrapMode> wrap_s;
+  BitField<2, 2, WrapMode> wrap_t;
+  BitField<4, 1, FilterMode> mag_filter;
+  BitField<5, 2, MipMode> min_mip;
+  BitField<7, 1, FilterMode> min_filter;
```

Comment:
This was split from what was originally `mag_filter : 1` and `min_filter : 3`.  I think the split makes sense (the code that used it treated the `min_filter` bit differently from the other 2, though `min_mip` is probably a bad name.

Reply:
I've changed it to `mipmap_filter` as that name is [already used in RenderState](https://github.com/dolphin-emu/dolphin/blob/9d0983c9c988f81c73c9a4ed61835ca9ac20db06/Source/Core/VideoCommon/RenderState.cpp#L221).

Some reverse-engineering I did found that Nintendo also treats it as a single 3-bit field, but the functions that set it (`gx::GXInitTexObjLOD` (8035fd94) and `JSystem::loadTexNo` (802eea04) in Super Mario Sunshine) convert the parameter using a 3-bit lookup table (`gx::GX2HWFiltConv` (8040cee0) and `JSystem::GX2HWFiltConv` (8040cc70)) with contents [0, 4, 1, 5, 2, 6, 0, 0].  This seems like unnecessary jank that doesn't need to be kept around.